Tools/LyTestTools/ly_test_tools/o3de/editor_test.py

```diff
@@ -467,7 +467,9 @@ def make_func(test_spec):
                         def result(self, request, workspace, editor, editor_test_data, launcher_platform):
                             # The runner must have filled the editor_test_data.results dict fixture for this test.
                             # Hitting this assert could mean if there was an error executing the runner
-                            assert test_spec.__name__ in editor_test_data.results, f"No run data for test: {test_spec.__name__}."
+                            assert test_spec.__name__ in editor_test_data.results, \
+                                f"No results found for {test_spec.__name__}. Test may not have ran due to the Editor " \
+                                f"shutting down. Check for issues in previous tests."
```

Comment:
This is an improvement, but can we do more before closing this issue? Looks like results is dict, and is not linear. Perhaps the data from batched results should instead be stored in a list, so that this (or another) summary can also suggest what test was the culprit for early exit/crashing?

Reply:
Moving from AssertionError to a typed exception may also help make it clear how/why the test is failing, could be similar to EditorTestNotReportedError

Reply:
I can change the exception type, but as per the first request I think it would require a larger feature outside this scope. The reporting that the parallel editor framework uses hooks into pytest's reporting so it's not trivial to provide a new summary